nixos/lib/test-driver/src/test_driver/machine/__init__.py

```diff
@@ -93,20 +93,25 @@ def make_command(args: list) -> str:
     return " ".join(map(shlex.quote, (map(str, args))))
 
 
-def retry(fn: Callable, timeout: int = 900) -> None:
-    """Call the given function repeatedly, with 1 second intervals,
-    until it returns True or a timeout is reached.
+def retry(fn: Callable, timeout: int = 900, interval: float = 1.0) -> None:
+    """Call the given function repeatedly, with a configurable interval between
+    retries, until it returns True or a timeout is reached.
+
+    Note that the timeout shown will include the time of the last attempted run.
     """
+    start_time = time.monotonic()
 
-    for _ in range(timeout):
+    while time.monotonic() - start_time < timeout:
         if fn(False):
             return
-        time.sleep(1)
+        time.sleep(interval)
 
-    if not fn(True):
```

Comment:
removing this last call with the bool param set to `True` (which basically means "this is the last call") changes the semantics.

Either we keep this call or we even remove the bool parameter because it becomes a broken promise with this change.

Reply:
Thank you for explaining this - it looked weird with the `return` before in the function when I first looked at it